src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -392,6 +392,395 @@ namespace OpenMS
       }
     }
 
+
+  /**
+   * @brief Calculates the sum of intensities for a range of elements.
+   * 
+   * @tparam Iterator The iterator type.
+   * @param begin The iterator pointing to the beginning of the range.
+   * @param end The iterator pointing to the end of the range.
+   * @return The sum of intensities.
+   * 
+   * @throws static assert fails if the iterator value type does not have a `getIntensity()` member function.
+   */
+struct SumIntensityReduction {
+
+  template<typename Iterator>
+  double operator()(Iterator begin, Iterator end) const {
```

Comment:
should be msexperiment::coordinatetype

Reply:
Or IntensityType I guess.

Reply:
Maybe I am braindead by now but I think this won't work. Internally, I have an iterator range of MSSpectrum::ConstIterator type (or equivalent).
This is fed to the reduction function (which could in e.g. use both m/z and intensity, e.g., for m/z weighted intensity or similar).

Reply:
Yeah but one will be double, the other float, so it would do needless promotions.

Reply:
Yes but you are not doing weighted sums. In this case you are summing intensities. So it should return a summed intensity (in case we switch types again).

Reply:
aah got it :)